java/arcs/android/storage/service/BindingContext.kt

```diff
@@ -74,81 +74,93 @@ class BindingContext(
     @VisibleForTesting
     val id = nextId.incrementAndGet()
 
-    /** The local [CoroutineContext]. */
+    /**
+     * The local [CoroutineContext], and a [CoroutineScope] that wraps it.
+     *
+     * All of the implemented AIDL methods will be immediately dispatched using this scope.
+     */
     private val job = Job(parentCoroutineContext[Job])
     private val coroutineContext =
         parentCoroutineContext + job + CoroutineName("BindingContext-$id")
+    private val scope = CoroutineScope(coroutineContext)
```

Comment:
We should really just pass this scope in as a constructor arg from the StorageService.

Reply:
Agreed. I almost did it, but got hung up because creating a scope in storage service is tricky since we override coroutineContext, so can't create a scope in the base class. But I think I can get around this for now with `lazy`, let me try that.

Reply:
Ok, follow-up: Just gave this a shot again. It touches on a number of things that I think are worth focusing on in an independent PR.